src/Sylius/Behat/Context/Api/Admin/ManagingExchangeRatesContext.php

```diff
@@ -0,0 +1,363 @@
+     */
+    public function iWantToEditThisExchangeRate(ExchangeRateInterface $exchangeRate): void
+    {
+        $this->sharedStorage->set('exchange_rate_id', $exchangeRate->getId());
+    }
+
+    /**
+     * @When /^I am editing (this exchange rate)$/
+     */
+    public function iAmEditingThisExchangeRate(ExchangeRateInterface $exchangeRate): void
+    {
+        $this->client->buildUpdateRequest('exchange_rates', $exchangeRate->getId());
+    }
+
+    /**
+     * @When I am browsing exchange rates of the store
+     * @When I browse exchange rates
+     * @When I browse exchange rates of the store
+     */
+    public function iBrowseExchangeRatesOfTheStore(): void
+    {
+        $this->client->index('exchange_rates');
+    }
+
+    /**
+     * @When /^I specify its ratio as (-?[0-9\.]+)$/
+     * @When I don't specify its ratio
+     */
+    public function iSpecifyItsRatioAs(?float $ratio = null): void
+    {
+        if($ratio !== null) {
+            $this->client->addRequestData('ratio', $ratio);
+        }
+    }
+
+    /**
+     * @When I choose :currencyCode as the source currency
+     */
+    public function iChooseAsTheSourceCurrency(string $currencyCode): void
+    {
+        $this->client->addRequestData('sourceCurrency', '/new-api/currencies/' . $currencyCode);
+    }
+
+    /**
+     * @When I choose :currencyCode as the target currency
+     */
+    public function iChooseAsTheTargetCurrency(string $currencyCode): void
+    {
+        $this->client->addRequestData('targetCurrency', '/new-api/currencies/' . $currencyCode);
+    }
+
+    /**
+     * @When I (try to) add it
+     */
+    public function iAddIt(): void
+    {
+        $this->client->create();
+    }
+
+    /**
+     * @When I change ratio to :ratio
+     */
+    public function iChangeRatioTo(string $ratio): void
+    {
+        $this->client->addRequestData('ratio', $ratio);
+    }
+
+    /**
+     * @When I save my changes
+     */
+    public function iSaveMyChanges(): void
+    {
+        $this->client->update();
+    }
+
+    /**
+     * @When I delete the exchange rate between :sourceCurrency and :targetCurrency
+     */
+    public function iDeleteTheExchangeRateBetweenAnd(CurrencyInterface $sourceCurrency, CurrencyInterface $targetCurrency) : void
+    {
+        /** @var ExchangeRateInterface */
+        $exchangeRate = $this->getExchangeRateBetweenCurrencies($sourceCurrency->getCode(), $targetCurrency->getCode());
+
+        $this->client->delete('exchange_rates', $exchangeRate->getId());
+    }
+
+    /**
+     * @When I choose :currency as a currency filter
+     */
+    public function iChooseCurrencyAsACurrencyFilter(CurrencyInterface $currency): void
+    {
+        $filters = $this->client->buildFilter('exchange_rates', 'sourceCurrency.code', $currency->getCode());
+        $filters = $this->client->buildFilter('exchange_rates', 'targetCurrency.code', $currency->getCode(), $filters);
+
+        $this->sharedStorage->set('filters', $filters);
```

Comment:
They should be saved in a request array in `ApiPlatformClient` instead of in shared storage